traitlets/traitlets.py

```diff
@@ -499,18 +499,33 @@ def init_default_value(self, obj):
         obj._trait_values[self.name] = value
         return value
 
+    def get_env_value(self):
+        """
+        Gets the value of any environment-variable named as `env` metadata.
+
+        :return:
+            the textual value of the environment variable, or None, meaning
+            that no `envvar` metadata is defined.
+        """
+        env_var = self.metadata.get('envvar')
+        return env_var and os.environ.get(env_var)
```

Comment:
>  that's a very common case.

It is also a very common case where setting an environment variable to an empty string is meant to be treated the same as. I have not personally encountered a case where setting an environment variable to an empty string should be treated differently to the environment variable not being set at all. Please make sure that env being an empty string is treated the same as the value not being defined. Also make sure that this value returns `Undefined` for no value being set, so that traits that allow None can still use this.